src/core/Sampler/Sampler.cpp

```diff
@@ -617,53 +641,44 @@ bool Sampler::renderNote( Note* pNote, unsigned nBufferSize, std::shared_ptr<Son
-			if ( Preferences::get_instance()->m_JackTrackOutputMode == Preferences::JackTrackOutputMode::postFader ) {
-				cost_track_L = 0.0;
-				cost_track_R = 0.0;
+		if ( bIsMutedForExport || pInstr->is_muted() || pSong->getIsMuted() ||
+			 pMainCompo->is_muted() || bIsMutedBecauseOfSolo) {	
+			fCost_L = 0.0;
+			fCost_R = 0.0;
+			if ( Preferences::get_instance()->m_JackTrackOutputMode ==
+				 Preferences::JackTrackOutputMode::postFader ) {
+				fCostTrack_L = 0.0;
+				fCostTrack_R = 0.0;
 			}
 
-		} else {	// Precompute some values...
+		} else {
+			float fMonoGain = 1.0;
 			if ( pInstr->get_apply_velocity() ) {
-				cost_L = cost_L * pNote->get_velocity();		// note velocity
-				cost_R = cost_R * pNote->get_velocity();		// note velocity
-			}
-
-
-			cost_L *= fPan_L;							// pan
-			cost_L = cost_L * fLayerGain;				// layer gain
-			cost_L = cost_L * pInstr->get_gain();		// instrument gain
-
-			cost_L = cost_L * pCompo->get_gain();		// Component gain
-			cost_L = cost_L * pMainCompo->get_volume(); // Component volument
-
-			cost_L = cost_L * pInstr->get_volume();		// instrument volume
-			if ( Preferences::get_instance()->m_JackTrackOutputMode == Preferences::JackTrackOutputMode::postFader ) {
-				cost_track_L = cost_L * 2;
+				fMonoGain *= pNote->get_velocity();	// note velocity
 			}
-			cost_L = cost_L * pSong->getVolume();	// song volume
-
-			cost_R *= fPan_R;							// pan
-			cost_R = cost_R * fLayerGain;				// layer gain
-			cost_R = cost_R * pInstr->get_gain();		// instrument gain
 
-			cost_R = cost_R * pCompo->get_gain();		// Component gain
-			cost_R = cost_R * pMainCompo->get_volume(); // Component volument
-
-			cost_R = cost_R * pInstr->get_volume();		// instrument volume
-			if ( Preferences::get_instance()->m_JackTrackOutputMode == Preferences::JackTrackOutputMode::postFader ) {
-				cost_track_R = cost_R * 2;
+			fMonoGain *= fLayerGain;				// layer gain
+			fMonoGain *= pInstr->get_gain();		// instrument gain
+			fMonoGain *= pCompo->get_gain();		// Component gain
+			fMonoGain *= pMainCompo->get_volume();	// Component volument
+			fMonoGain *= pInstr->get_volume();		// instrument volume
+			fMonoGain *= pSong->getVolume();		// song volume
+
+			fCost_L = fMonoGain * fPan_L;			// pan
+			fCost_R = fMonoGain * fPan_R;			// pan
+			if ( Preferences::get_instance()->m_JackTrackOutputMode ==
+				 Preferences::JackTrackOutputMode::postFader ) {
+				fCostTrack_R = fCost_R * 2;
+				fCostTrack_L = fCost_L * 2;
 			}
-			cost_R = cost_R * pSong->getVolume();	// song pan
 		}
 
 		// direct track outputs only use velocity
-		if ( Preferences::get_instance()->m_JackTrackOutputMode == Preferences::JackTrackOutputMode::preFader ) {
-			cost_track_L = cost_track_L * pNote->get_velocity();
-			cost_track_L = cost_track_L * fLayerGain;
-			cost_track_R = cost_track_L;
+		if ( Preferences::get_instance()->m_JackTrackOutputMode ==
+			 Preferences::JackTrackOutputMode::preFader ) {
+			fCostTrack_L *= pNote->get_velocity();
```

Comment:
Refactoring the gain contributions made more obvious that note velocity is always applied in per track outputs in PreFader mode  but only applied in PostFader mode in case the "apply velocity" checkbox in the Instrument Editor was checked. This does not seem right.

Reply:
In addition, PreFader gain does not take the note pan into account.